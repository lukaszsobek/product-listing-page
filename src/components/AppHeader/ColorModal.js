import React, { Component } from "react";
import { connect } from "react-redux";

import { toggleModal } from "../../actions";

const DropDown = props => {
    if(!props.isOpen) {
        return null;
    }

    return(
        <div className="modal--color__data">
            something
        </div>
    );
}

class ColorModal extends Component {

    render() {
        const { isOpen, toggleModal } = this.props;
        return (
            <div className="modal--color">
                <div
                    className="modal--color__title"
                    onClick={toggleModal}
                >
                    Colors
                </div>
                <DropDown isOpen={isOpen}/>
            </div>
        );
    }
};

const mapStateToProps = state => ({
    isOpen: state.modalState.colorModalVisible
});

const mapDispatchToProps = dispatch => ({
    toggleModal: () => dispatch(toggleModal("ColorModal"))
});

export default connect(mapStateToProps, mapDispatchToProps)(ColorModal);