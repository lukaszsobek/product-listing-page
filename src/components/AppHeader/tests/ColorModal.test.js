import { ColorModal, DropDown } from "../ColorModal";

describe("ColorModal", () => {

    it("renders ColorModal",() => {
        const props = {
            data: [1,2], 
            isOpen: false, 
            updatePage: () => null, 
            toggleModal: () => null
        }
        expect(new ColorModal(props).render()).toMatchSnapshot();
    });

    it("returns null if no DropDown",() => {
        const props = {}
        expect(DropDown(props)).toBe(null);
    });

    it("renders DropDown",() => {
        const props = {
            isOpen: true,
            data: [1,2]
        }
        expect(DropDown(props)).toMatchSnapshot();
    });
});