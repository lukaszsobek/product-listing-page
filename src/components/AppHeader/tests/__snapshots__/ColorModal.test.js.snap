// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorModal renders ColorModal 1`] = `
<div
  className="modal--color"
>
  <div
    className="modal--color__title"
    onClick={[Function]}
  >
    Colors
  </div>
  <DropDown
    data={
      Array [
        1,
        2,
      ]
    }
    isOpen={false}
    updatePage={[Function]}
  />
</div>
`;

exports[`ColorModal renders DropDown 1`] = `
<div
  className="modal--color__data"
>
  <div
    onClick={[Function]}
  >
    1
  </div>
  <div
    onClick={[Function]}
  >
    2
  </div>
  <div>
    ..clear
  </div>
</div>
`;
